node-operators/run-aut/index.md

```diff
@@ -22,26 +22,77 @@ To connect to a network and sync, get the genesis and bootnode files if needed,
+
+	```bash
+	./autonity/build/bin/autonity genAutonityKeys --writeaddress ./keystore/myAutonityKeysFile
+	```
+
+	You should see something like this output to your terminal:
+
+	```bash
+	Node address: 0x0f083e047725bF1dFf3efe110430B98AAac55986
+	Node public key: 0xee0f1195d19b422cd6fc69b060853b7c21094a84e66a225b549afafe7ba44996f85015d6a7f19d32e1e978db26db4083c8db8e153ddea6f1aa4c65608374c868
+	Consensus public key: 0x8aa83a28e235072ffdae48ff01ccc46e2b8d9dc16df9b6ff87ffa5ff6d8f90a2852649a60563237cd66a256f60a92e69
+	```
+
+::: {.callout-important title="Important" collapse="false"}
+Remember to backup your `autonitykeys` file! Copy it to a safe location!
+:::
+
+::: {.callout-important title="Important" collapse="false"}
+Copy the output - you will need the output if you intend to register your validator as a node.
+
+It is *not* possible to inspect the keyfile to retrieve the `Consensus public key` which you MUST provide as part of the validator registration information.
+:::
+
+
+4. Start autonity:
 
     ``` bash
     autonity \
-        --datadir ./autonity-chaindata  \
+        --datadir ./autonity-chaindata \
+        --autonitykeys ./<PATH_TO_AUTONITYKEYS_FILE> \
         --piccadilly  \
         --http  \
         --http.addr 0.0.0.0 \
-        --http.api aut,eth,net,txpool,web3,admin  \
-        --http.vhosts \* \
+        --http.api aut,eth,net,txpool,web3,admin \
+        --http.vhosts "*" \
         --ws  \
         --ws.addr 0.0.0.0 \
         --ws.api aut,eth,net,txpool,web3,admin  \
-        --nat extip:<IP_ADDRESS>
+        --nat extip:<IP_ADDRESS> \
+        --consensus.nat <CONSENSUS_NAT> \
+        --consensus.port <CONSENSUS_PORT_NUMBER> \
+        ;
     ```
 
    where:
 
+   - `<PATH_TO_AUTONITYKEYS_FILE>` is the path to the `autonitykeys` file generated in Step 2.
    - `<IP_ADDRESS>` is the node's host IP Address, which can be determined with `curl ifconfig.me`.
-   - `--piccadilly` specifies that the node will use the Piccadilly tesnet.  For other tesnets, use the appropriate flag (for example, `--bakerloo`).
+   - (Optional) `<CONSENSUS_NAT>` specify the NAT port mapping for the consensus channel (one of "any", "none", "upnp", "pmp", "extip:<IP>") if the default value "any" is not to be used.
+   - (Optional) `<CONSENSUS_PORT_NUMBER>` specify the network listening port for the consensus channel if the default port "20203" is not to be used.
+   - `--piccadilly` specifies that the node will connect to the Piccadilly testnet.  For other testnets, use the appropriate flag (for example, `--bakerloo`).
+
+::: {.callout-warning title="Warning!" collapse="false"}
+You *must* specify the `--autonitykeys` command option.
+
+If it is *not* specified, then on starting AGC will automatically generate a standard Ethereum `nodekey` file  by default within the `autonity` subfolder of the `--datadir` specified when running the node. 
```

Comment:
If it is not specified, AGC will auto generate a brand new autonityKeys file which contains the node key and consensus key for you in that subfolder. The auto generated keys file will be used by AGC.

Reply:
Thanks, corrected! See commit e5e1b19d1dfd65cb8201f6f4256ebd5099c9539d